File: hfsfolder-python/model/Folder.py
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 08/04/2019
# * Time: 14:04
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
#
from model.PreFile import PreFile

class Folder(PreFile):
	""" <summary>
	 Description of Folder.
	 </summary>
	"""
	def __init__(self, preFile):
		self.setName(preFile.getName())
		self.setSize(preFile.getSize())
		self.setModified(preFile.getModified())
		self.setAttributes(preFile.getAttributes())
		self.setFormatedSize(preFile.getFormatedSize())
		self.setFormatedModified(preFile.getFormatedModified())
	
	def getCode(self):
		return self._code

	def setCode(self, code):
		self._code = code

	def getOrder(self):
		return self._order

	def setOrder(self, order):
		self._order = order

	def getParentCodFolder(self):
		return self._parentCodFolder

	def setParentCodFolder(self, parentCodFolder):
		self._parentCodFolder = parentCodFolder

	def getFolderType(self):
		return self._folderType

	def setFolderType(self, folderType):
		self._folderType = folderType

	def getPath(self):
		return self._path

	def setPath(self, path):
		self._path = path

	def getParentName(self):
		return self._parentName

	def setParentName(self, parentName):
		self._parentName = parentName

	def getParentPath(self):
		return self._parentPath

	def setParentPath(self, parentPath):
		self._parentPath = parentPath

	def limparDados(self):
		self.limparDados()
		self._code = 0
		self._order = 0
		self._parentCodFolder = 0
		self._folderType = 'A'
		self._path = ""
		self._parentName = ""
		self._parentPath = ""

	def toString(self):
		return f"Folder [codigo={self._code}, ordem={self._order}, codDirPai={self._parentCodFolder}, " \
			f"tipo={self._folderType}, caminho={self._path}, nomePai={self._parentName}, " \
			f"caminhoPai={self._parentPath}]"

	def toInsert(self, naba):
		return f"INSERT INTO Diretorios(aba, cod, ordem, nome, tam, tipo, modificado, atributos, coddirpai, caminho) " \
			f"VALUES({naba.ToString()},{self._code},{self._order},'{self.getName()}'," \
			f"{self.getSize()},'{self._folderType}','{self.getFormatedModified()}','" \
			f"{self.getAttributes()}',{self._parentCodFolder},'{self.getPath()}');"

	def toCVS(self):
		return f"{self._code};{self._order};{self.getName()};{self.getSize()};" \
			f"{self._folderType};{self.getFormatedModified()};{self.getAttributes()};" \
			f"{self._parentCodFolder};{self.getPath()}"

	def toJSON(self):
		str = f'{{\n  "name" : "{self.getName()}",'
		str += f'\n  "size" : {self.getSize()},'
		str += f'\n  "modified" : {self.getModified()},'
		str += f'\n  "attributes" : "{self.getAttributes()}",'
		str += f'\n  "formatedSize" : {self.getFormatedSize()},'
		str += f'\n  "formatedModified" : "{self.getFormatedModified()}",'
		str += f'\n  "code" : {self._code},'
		str += f'\n  "order" : {self._order},'
		str += f'\n  "parentCodFolder" : {self._parentCodFolder},'
		str += f'\n  "folderType" : "{self._folderType}",'
		str += f'\n  "path" : "{self._path}",'
		str += f'\n  "parentName" : "{self._parentName}",'
		str += f'\n  "parentPath" : "{self._parentPath}",'
		str += f'\n  "originalPath" : "{self.getOriginalPath()}"\n}}'

		return str

File: hfsfolder-ruby/model/Folder.rb
# * User: henrique.souza
# * Date: 08/04/2019
# * Time: 14:04

require_relative 'PreFile'

class Folder < PreFile

  def initialize(preFile)
    self.setName(preFile.getName)
    self.setSize(preFile.getSize)
    self.setModified(preFile.getModified)
    self.setAttributes(preFile.getAttributes)
    self.setFormatedSize(preFile.getFormatedSize)
    self.setFormatedModified(preFile.getFormatedModified)
    self.separatorChar = '/'
  end

  def getCode
    @code
  end

  def setCode(code)
    @code = code
  end

  def getOrder
    @order
  end

  def setOrder(order)
    @order = order
  end

  def getParentCodFolder
    @parentCodFolder
  end

  def setParentCodFolder(parentCodFolder)
    @parentCodFolder = parentCodFolder
  end

  def getFolderType
    @folderType
  end

  def setFolderType(folderType)
    @folderType = folderType
  end

  def getPath
    @path
  end

  def setPath(path)
    @path = path
  end

  def getParentName
    @parentName
  end

  def setParentName(parentName)
    @parentName = parentName
  end

  def getParentPath
    @parentPath
  end

  def setParentPath(parentPath)
    @parentPath = parentPath
  end

  def limparDados
    self.limparDados
    @code = 0
    @order = 0
    @parentCodFolder = 0
    @folderType = 'A'
    @path = ""
    @parentName = ""
    @parentPath = ""
  end

  def toString
    "Folder [codigo=" + @code.to_s + ", ordem=" + @order.to_s + ", codDirPai=" + @parentCodFolder.to_s + ", tipo=" +
        @folderType + ", caminho=" + @path + ", nomePai=" + @parentName + ", caminhoPai=" + @parentPath + "]"
  end

  def toInsert(naba)
    "INSERT INTO Diretorios(aba, cod, ordem, nome, tam, tipo, modificado, atributos, coddirpai, caminho) VALUES(" +
        naba.to_s + "," + @code.to_s + "," + @order.to_s + ",'" + getName + "'," + getSize.to_s + ",'" + @folderType + "','" +
        getFormatedModified + "','" + getAttributes + "'," + @parentCodFolder.to_s + ",'" + getPath + "');"
  end

  def toCVS
    @code.to_s + ";" + @order.to_s + ";" + getName + ";" + getSize.to_s + ";" + @folderType + ";" + getFormatedModified +
        ";" + getAttributes + ";" + @parentCodFolder.to_s + ";" + getPath
  end

  def toJSON
    "{\n  \"name\" : \"" + getName + "\"," + "\n  \"size\" : " + getSize.to_s + "," +
        "\n  \"modified\" : " + getModified.to_s + "," + "\n  \"attributes\" : \"" + getAttributes + "\"," +
        "\n  \"formatedSize\" : \"" + getFormatedSize + "\"," +
        "\n  \"formatedModified\" : \"" + getFormatedModified + "\"," + "\n  \"code\" : " + @code.to_s + "," +
        "\n  \"order\" : " + @order.to_s + "," + "\n  \"parentCodFolder\" : " + @parentCodFolder.to_s + "," +
        "\n  \"folderType\" : \"" + @folderType + "\"," + "\n  \"path\" : \"" + @path + "\"," +
        "\n  \"parentName\" : \"" + @parentName + "\"," + "\n  \"parentPath\" : \"" + @parentPath + "\"," +
        "\n  \"originalPath\" : \"" + getOriginalPath + "\"\n}"
  end

  def separatorChar
    @separatorChar
  end

  def separatorChar=(value)
    @separatorChar = value
  end

end
